presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSource.java

```diff
@@ -147,6 +148,9 @@ else if (type instanceof DecimalType || type instanceof DoubleType) {
             else if (type instanceof TimestampType) {
                 type.writeLong(blockBuilder, parseLong(value));
             }
+            else if (type instanceof DateType) {
+                blockBuilder.writeInt(parseDouble(value).intValue());
```

Comment:
Why parseDouble and then convert that to int instead of directly parsing to int ?

Reply:
as for some queries like max/min, pinot will return a double value in the format of "16453.0000". So we need to parse a double first, then convert to int.